providers/sensu_datacenter.rb

```diff
@@ -0,0 +1,14 @@
+action :create do
+  datacenter = {}
+  datacenter[:name] = new_resource.name unless new_resource.name.nil?
+  datacenter[:host] = new_resource.host
+  datacenter[:port] = new_resource.port
+  datacenter[:ssl] = new_resource.ssl
+  datacenter[:insecure] = new_resource.insecure
+  datacenter[:path] = new_resource.path unless new_resource.path.nil?
+  datacenter[:timeout] = new_resource.timeout
+  datacenter[:user] = new_resource.user unless new_resource.user.nil?
+  datacenter[:pass] = new_resource.pass unless new_resource.pass.nil?
+
+  node.default["uchiwa"]["datacenters"] << datacenter
```

Comment:
Shouldn't this first check for the existence of a datacenter with the same name, then compare hash objects before adding to the node attribute array? Can then notify resource updated if it's added.